src/index.js

```diff
@@ -1,21 +1,55 @@
-import ReactDOM from "react-dom";
-import React from "react";
+/**
+ * External dependencies
+ */
+import ReactDOM from 'react-dom';
+import React from 'react';
+import head from 'lodash/head';
 import { DropZoneProvider, DropZone } from '@wordpress/components';
 import { withState } from '@wordpress/compose';
+import apiFetch from '@wordpress/api-fetch';
+
+const validFileTypes = [ 'text/xml' ];
+const isValidFileType = type => validFileTypes.includes( type );
 
 const MyDropZone = withState( {
 	hasDropped: false,
-} )( ( { hasDropped, setState } ) => (
-	<DropZoneProvider>
-		<div>
-			{ hasDropped ? 'Dropped!' : 'Drop something here' }
-			<DropZone 
-				onFilesDrop={ () => setState( { hasDropped: true } ) }
-				onHTMLDrop={ () => setState( { hasDropped: true } )  }
-				onDrop={ () => setState( { hasDropped: true } ) } 
-			/>
-		</div>
-	</DropZoneProvider>
-) );
+} )( ( { hasDropped, setState } ) => {
+	const doIt = async ( files = [] ) => {
```

Comment:
At this point we might want to convert from a functional component to a Component class extension so as to not redefine `doIt` on each re-render. That said, while we're still on PoC turf we don't really need to worry so much about these issues so I'd be happy to address this further down the line :)

Reply:
I think we'll want to promote it to a proper action with a proper name -- calling it `doIt` was sort of an admission that we need to figure out the data flow stuff soon-ish :)